openedx/core/djangoapps/content_tagging/handlers.py

```diff
@@ -74,3 +89,39 @@ def delete_tag_xblock(**kwargs):
         delete_course_tags.delay(str(xblock_info.usage_key.course_key))
 
     delete_xblock_tags.delay(str(xblock_info.usage_key))
+
+
+@receiver(LIBRARY_BLOCK_CREATED)
+@receiver(LIBRARY_BLOCK_UPDATED)
+def auto_tag_library_block(**kwargs):
+    """
+    Automatically tag Library Blocks based on metadata
+    """
+    library_block_data = kwargs.get("library_block", None)
+    if not library_block_data or not isinstance(library_block_data, LibraryBlockData):
+        log.error("Received null or incorrect data for event")
+        return
+
+    if not CONTENT_TAGGING_AUTO.is_enabled():
+        return
```

Comment:
nit: check the flag before gathering library_block data
```suggestion
    if not CONTENT_TAGGING_AUTO.is_enabled():
        return
    library_block_data = kwargs.get("library_block", None)
    if not library_block_data or not isinstance(library_block_data, LibraryBlockData):
        log.error("Received null or incorrect data for event")
        return
```

Reply:
Updated the commit with this change.